addon/utils/utils.js

```diff
@@ -7,18 +7,44 @@ import Ember from 'ember';
 
 const DS = requireModule('ember-data');
 
+const {
+  get
+} = Ember;
+
 export function requireModule(module) {
   return self.requirejs.has(module) ? self.require(module).default : undefined;
 }
 
-export function unwrapString(input) {
-  if (input && input instanceof Ember.Handlebars.SafeString) {
-    return input.toString();
+export function unwrapString(s) {
+  if (s && s instanceof Ember.Handlebars.SafeString) {
+    return s.toString();
+  }
+
+  return s;
+}
+
+export function unwrapProxy(o) {
+  if (o && (o instanceof Ember.ObjectProxy || o instanceof Ember.ArrayProxy)) {
```

Comment:
You could simplify this to cover anything using the mixin if you wanted. 

```
if (o && (Ember.PromiseProxyMixin.detect(o))
```

Covers a bigger range of promises with a content. Just a thought. I use it in from https://github.com/unchartedcode/ember-promise-tools/blob/master/addon/utils/get-promise-content.js